cirq-web/cirq_web/bloch_sphere/bloch_sphere.py

```diff
@@ -0,0 +1,158 @@
+        """
+
+        super().__init__('cirq_ts/dist/bloch_sphere.bundle.js')
+
+        self.sphere_json = self._convert_sphere_input(sphere_radius)
+        self.bloch_vector = (
+            self._create_random_vector() if random else self._create_vector(state_vector)
+        )
+        self.vector_json = self._serialize_vector(*self.bloch_vector, sphere_radius)
+
+        # Generate a unique UUID for every instance of a Bloch sphere.
+        # This helps with adding visualizations to scenes, etc.
+        self.id = str(uuid.uuid1())
+
+    def _repr_html_(self):
+        """Allows the object's html to be easily displayed in a notebook
+        by using the display() method.
+
+        If display() is called from the command line, [INSERT HERE]
+        """
+
+        bundle_script = super().get_bundle_script()
+        return f"""
+        <meta charset="UTF-8">
+        <div id="{self.id}"></div>
+        {bundle_script}
+        <script>
+        renderBlochSphere('{self.sphere_json}', '{self.id}').addVector('{self.vector_json}');
+        </script>
+        """
+
+    def get_id(self):
+        return self.id
+
+    def generate_html_file(
+        self,
+        output_directory: str = './',
+        file_name: str = 'bloch_sphere.html',
+        open_in_browser: bool = False,
+    ) -> Union[PosixPath, WindowsPath]:
+        """Generates a portable HTML file of the bloch sphere that
+        can be run anywhere. Prints out the absolute path of the file to the console.
+
+        Args:
+            output_directory: the directory in which the output file will be
+            generated. The default is the current directory ('./')
+
+            file_name: the name of the output file. Default is 'bloch_sphere'
+
+            open: if True, opens the newly generated file automatically in the browser.
+
+        Returns:
+            The path of the HTML file in either PosixPath or WindowsPath form, depending on the
+            operating system.
+
+        For now, if ran in a notebook, this function just returns. Support for downloading
+        the HTML file via the browser can be added later.
+        """
+
+        template_div = f"""
+        <meta charset="UTF-8">
+        <div id="{self.id}"></div>
+        """
+
+        template_script = f"""
+        <script>
+        renderBlochSphere('{self.sphere_json}', '{self.id}').addVector('{self.vector_json}');
+        </script>
+        """
+
+        bundle_script = super().get_bundle_script()
+        contents = template_div + bundle_script + template_script
+        path_of_html_file = widget.write_output_file(output_directory, file_name, contents)
+
+        if open_in_browser:
+            webbrowser.open(str(path_of_html_file), new=2)  # 2 opens in a new tab if possible
+
+        return path_of_html_file
+
+    def _convert_sphere_input(self, radius: int) -> str:
+        if radius <= 0:
+            raise (BaseException('You must input a positive radius for the sphere'))
```

Comment:
We never raise `BaseException`. Typically `ValueError` is raised when the provided arguments fail validation.

Reply:
Also, outer parenthesis are not required.